### PR TITLE
linter: added storage of default values for params of functions and methods

### PR DESCRIPTION
**Pull request by i582**

### CODE REVIEW

File: src/linter/cache_test.go [hunk 1]
```diff
@@ -101,6 +101,18 @@ function as_shape($x) {
...
 }
 
+/** With default */
+function withDefault($x = 100) {
+  return $x;
+}
+
+class WithDefaultMethod {
+  /** With default */
+  public function withDefault($x = 100) {
```
Comment (quasilyte) [inline src/linter/cache_test.go:111]: Use different type of the default param here. String, for example.

File: src/meta/metainfo.go [hunk 2]
```diff
@@ -312,9 +312,10 @@ func (i *info) AddToGlobalScopeNonLocked(filename string, sc *Scope) {
 }
 
 type FuncParam struct {
-	IsRef bool
-	Name  string
-	Typ   TypesMap
+	IsRef   bool
+	Name    string
+	Typ     TypesMap
+	Default *ConstValue
```
Comment (quasilyte) [inline src/meta/metainfo.go:318]: Why can't it be a non-pointer type?
Zero value ConstValue has Type=Undefined, which is good enough to detect "no default value" case.

File: src/linter/cache_test.go (hunk 1 above)
Comment (i582) [reply src/linter/cache_test.go]: Done.

File: src/meta/metainfo.go (hunk 2 above)
Comment (i582) [reply src/meta/metainfo.go]: Good idea. Done.

**Comment by i582** [comment]
Let's leave until better times.

**Pull request closed without merging**